### PR TITLE
Add autogenerate sdk references

### LINKED ISSUE #991: Create workflow to automatically create API references for SDK
**Is your feature request related to a problem? Please describe.**

API references should be generated automatically on wiki builds. They should not be checked in.

**Describe the solution you'd like**

We probably need to do a shallow clone of the repo and than run the needed commands to create the needed docs

**Describe alternatives you've considered**

I guess we could also check if our tools allow using remote code, but very unlikely that all of them do...

### PR DESCRIPTION
**Pull request by Dr-Electron**
# Description of change

Created a workflow to autogenerate the API references for the SDK docs. Not the prettiest solution, but I guess it works.
Also removed a flag from the nodejs generation to document all public properties (I hope 😅): https://github.com/iotaledger/iota-sdk/issues/634

## Links to any relevant issues

Fixes #991 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Dr-Electron** [comment]
We should probably add the typedoc command to the SDK repo as it was before. So everything related to references is always configures on their side

**Review by lucas-tortora — APPROVED** [review]

**Review by lucas-tortora — APPROVED** [review]

**Comment by Dr-Electron** [comment]
Does somebody know why the previews don't have the WASM reference? If I navigate to that page manually, it is shown as if it wouldn't be part of the sidebar 🤔 
@jlvandenhout @lucas-tortora

File: next/docs/iota-sdk/sidebars.js [hunk 1]
```diff
@@ -9,6 +9,75 @@
  Create as many sidebars as you want.
  */
 
+const fs = require('fs');
+const path = require('path');
```
Comment (Dr-Electron) [inline next/docs/iota-sdk/sidebars.js:13]:
```suggestion
```

**Review by lucas-tortora — APPROVED** [review]

**Pull request merged**